src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,42 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static addAnimal(animal) {
+    Animal.alive.push(animal);
+  }
+
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.addAnimal(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(herb) {
+    if (herb instanceof Herbivore && !herb.hidden) {
+      herb.health -= 50;
+
+      if (herb.health === 0) {
+        const removedAnimal = Animal.alive.indexOf(herb);
+
+        Animal.alive.splice(removedAnimal, 1);
```

Comment:
it will be better to use `filter` method

Reply:
@vlad-demchuk  Done